### PR TITLE
Change sanitization check for `ExportDataAsCode`

### PR DESCRIPTION
**Pull request by lauchimoon**
I opted to use `isalnum` function since it should handle most cases. It cannot however handle cases of files beginning with numbers.

### CODE REVIEW

**Comment by orcmid** [comment]
It is good to use isalnum.  Please see https://github.com/raysan5/raylib/pull/3814#issuecomment-1961645534 and subsequent comments for covering all cases and providing a utility function for this.  I still have no idea why capitalization is needed.  I think it is a misplaced application of the preprocessor identifier capitalization practice.  That should be no issue in this use-case.

File: src/utils.c [hunk 1]
```diff
@@ -48,6 +48,7 @@
 #include <stdio.h>                      // Required for: FILE, fopen(), fseek(), ftell(), fread(), fwrite(), fprintf(), vprintf(), fclose()
 #include <stdarg.h>                     // Required for: va_list, va_start(), va_end()
 #include <string.h>                     // Required for: strcpy(), strcat()
+#include <ctype.h>                      // Required for: isalnum()
```
Comment (raysan5) [inline src/utils.c:51]: In raylib I try to minimize standard library dependency, I prefer to avoid it.

Comment (lauchimoon) [reply src/utils.c:51]: Alright, I'll find a workaround, thanks

Comment (raysan5) [reply src/utils.c:51]: I can review it if you want.

**Pull request merged**